presto-router/src/test/java/com/facebook/presto/router/TestRouterAuthentication.java

```diff
@@ -0,0 +1,228 @@
+                .setOptionalConfigurationProperty("http-server.https.keystore.path", keystorePath.toAbsolutePath().toString())
+                .setOptionalConfigurationProperty("http-server.https.keystore.key", jksPassword)
+                .initialize();
+        injector.getInstance(RouterPluginManager.class).loadPlugins();
+
+        // Add auth on the fly to avoid needing to write etc/password-authenticator.properties
+        PasswordAuthenticatorManager passwordAuthenticatorManager = injector.getInstance(PasswordAuthenticatorManager.class);
+        PasswordAuthenticatorFactory authFactory = new FileAuthenticatorFactory();
+        passwordAuthenticatorManager.addPasswordAuthenticatorFactory(authFactory);
+        passwordAuthenticatorManager.setRequired();
+        passwordAuthenticatorManager.loadPasswordAuthenticator();
+
+        lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        httpServerInfo = injector.getInstance(HttpServerInfo.class);
+        httpClient = new JettyHttpClient(
+                new HttpClientConfig()
+                        .setTrustStorePath(truststorePath.toAbsolutePath().toString())
+                        .setTrustStorePassword(jksPassword));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+    {
+        if (authenticatorPropertiesFile != null) {
+            authenticatorPropertiesFile.delete();
+        }
+        if (lifeCycleManager != null) {
+            lifeCycleManager.stop();
+        }
+    }
+
+    @DataProvider(name = "routerEndpoints")
+    public Object[][] routerEndpoints()
+    {
+        return new Object[][] {
+                {"GET", "/"},
+                {"GET", "/v1/info"},
+                {"GET", "/v1/cluster"},
+                {"POST", "/v1/statement"},
+                // TODO: The UI is intentionally left out because the airlift incorrectly configures
+                //  some static resources
+                // {"GET", "/ui/"}
+        };
+    }
+
+    @Test(dataProvider = "routerEndpoints")
+    public void testEndpointWithoutAuthentication(String method, String endpoint)
+            throws Exception
+    {
+        Request request = Request.builder()
+                .setUri(UriBuilder.fromUri(httpServerInfo.getHttpsUri()).path(endpoint).build())
+                .setMethod(method)
+                .build();
+        httpClient.execute(request, new ResponseHandler<Void, Exception>()
+        {
+            @Override
+            public Void handleException(Request request, Exception exception)
+                    throws Exception
+            {
+                throw exception;
+            }
+
+            @Override
+            public Void handle(Request request, Response response)
+                    throws Exception
+            {
+                assertEquals(response.getStatusCode(), 401, format("request to %s should not have been successful. Expected 401, got: %d%n%s",
```

Comment:
```suggestion
                assertEquals(response.getStatusCode(), 401, format("request to %s should not have been successful. Expected 401, got: %d - %n - %s",
```

Reply:
This will put the first line of the HTTP response on the same line as the error message which can make reading the HTTP body in the error more difficult. I think it is ok to have it on a separate line

Reply:
ahh understood, I forgot `%n` is next line.